geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaInputFormat.scala

```diff
@@ -155,13 +157,16 @@ class GeoMesaInputFormat extends InputFormat[Text, SimpleFeature] {
    * number of shards in the schema as a proxy for number of tservers. We also know that each range will
    * only have a single location, because we always set autoAdjustRanges in the configure method above.
    */
-  override def getSplits(context: JobContext): java.util.List[InputSplit] = {
+  override def getSplits(context: JobContext) : java.util.List[InputSplit] = {
     init(context.getConfiguration)
     val accumuloSplits = delegate.getSplits(context)
-    // try to create 2 mappers per node - account for case where there are less splits than shards
-    val groupSize = Math.max(numShards * 2, accumuloSplits.length / (numShards * 2))
+    var groupSize = Math.max(numShards * 2, accumuloSplits.length / (numShards * 2)) // preserve original behavior.
+    if (desiredSplitCount != -1)
+    {
+      groupSize = Math.max(1, accumuloSplits.length / desiredSplitCount)
+    }
 
-    // We know each range will only have a single location because of autoAdjustRanges
+   // We know each range will only have a single location because of autoAdjustRanges
```

Comment:
hmm, it might. i believe it also merges ranges up though, e.g. if you have a range that spans 2 tservers it will split it into 2 ranges. my memory on exactly how it works is hazy...